File: api/services/reproduction.go
package services

import (
	"github.com/ONSBR/Plataforma-Deployer/sdk/eventmanager"
	"github.com/ONSBR/Plataforma-EventManager/domain"
)

type ReproductionService struct {
}

func (rep *ReproductionService) Reproduce(instanceID, owner string) error {
	e := domain.Event{
		Name: "system.events.reproduction.request",
		Reproduction: map[string]interface{}{
			"instanceId": instanceID,
			"owner":      owner,
		},
	}
	return eventmanager.Push(&e)
}

func NewReproductionService() *ReproductionService {
	return new(ReproductionService)
}

File: frontend/ui/src/Services/api/reproduction.js
import ApiService from './index'
import Axios from 'axios'

class ReproductionService extends ApiService {
    constructor(){
        super()
    }
    reproduce(instanceId, owner){
        return Axios.post(this.url("reproduction/"+instanceId,{owner}))
    }
}

export default ReproductionService